trial-tomland/src/Trial/Tomland.hs

```diff
@@ -0,0 +1,73 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Trial helpers for @tomland@.
+-}
+
+module Trial.Tomland
+    ( trialCodec
+    , trialTextCodec
+    , taggedTrialCodec
+    , taggedTrialTextCodec
+    ) where
+
+import Data.Text (Text)
+import Toml (Key, TomlCodec)
+import Trial (TaggedTrial, Trial (..), fiasco, maybeToTrial, trialToMaybe, unTag, withTag)
+
+import qualified Toml
+
+
+{- | 'TomlCodec' for 'Trial' that adds a given event @e@ if a given
+codec fails.
+
+@since 0.0.0.0
+-}
+trialCodec :: e -> TomlCodec a -> TomlCodec (Trial e a)
+trialCodec e = Toml.dimap trialToMaybe (maybeToTrial e) . Toml.dioptional
+
+{- | 'TomlCodec' for 'Trial' that adds an informative message if a
+given codec fails.
+
+@since 0.0.0.0
+-}
+trialTextCodec :: forall a . (Key -> TomlCodec a) -> Key -> TomlCodec (Trial Text a)
```

Comment:
Maybe instead of `Text` we can make it as in `Labels`: `Semigroup e, IsString e`?